src/leader_scheduler.rs

```diff
@@ -322,6 +335,77 @@ impl LeaderScheduler {
     }
 }
 
+pub fn is_leader_rotation_height(
+    height: u64,
+    bootstrap_height: u64,
+    leader_rotation_interval: u64,
+) -> bool {
+    if height < bootstrap_height {
+        return false;
+    }
+
+    if (height - bootstrap_height) % leader_rotation_interval == 0 {
+        return true;
+    }
+
+    false
+}
+
+pub fn entries_until_next_leader_rotation(
+    height: u64,
+    bootstrap_height: u64,
+    leader_rotation_interval: u64,
+) -> u64 {
+    if height < bootstrap_height {
+        bootstrap_height - height
+    } else {
+        leader_rotation_interval - ((height - bootstrap_height) % leader_rotation_interval)
+    }
+}
+
+// Remove all candiates for leader selection from the active set by clearing the bank,
+// and then set a single new candidate who will be eligible starting at height = vote_height
+// by adding one new account to the bank
+#[cfg(test)]
+pub fn set_new_leader(bank: &Bank, leader_scheduler: &mut LeaderScheduler, vote_height: u64) {
+    // Set the scheduled next leader to some other node
+    let new_leader_keypair = Keypair::new();
+    let new_leader_id = new_leader_keypair.pubkey();
+    leader_scheduler.push_vote(new_leader_id, vote_height);
+    let dummy_id = Keypair::new().pubkey();
+    let new_account = Account::new(1, 10, dummy_id.clone());
+
+    // Remove the previous acounts from the active set
+    let mut accounts = bank.accounts().write().unwrap();
+    accounts.clear();
+    accounts.insert(new_leader_id, new_account);
+}
+
+// Create two entries so that the node with keypair == active_keypair
+// is in the active set for leader selection:
+// 1) Give him nonzero number of tokens,
+// 2) A vote from the validator
+pub fn make_active_set_entries(
```

Comment:
this also a test function?

Reply:
yup! but it's also used by the integration tests in multinode.rs, so I couldn't mark it with the "test" annotation